tests/ReservationTest.php

```diff
@@ -0,0 +1,64 @@
+<?php
+namespace Tests;
+
+use App\Models\Reservation;
+use App\Models\User;
+use App\Resources\ReservationResource;
+use App\Resources\TermResource;
+
+/**
+ * - /reservation/{id}:get
+ * - /reservation/{id}/check:get
+ * - /reservation/search:get
+ */
+class ReservationTest extends TestCase {
+
+    /**
+     * /reservation/{id}:get
+     */
+    public function testShow() {
+        $user = User::factory()->permission('reservation')->create();
+        $reservation = Reservation::factory()->create();
+        $this->actingAs($user)->get("/reservations/{$reservation->id}");
+        $this->assertResponseOk();
+        $this->assertJson($this->response->getContent());
+        $this->seeJsonEquals([
+            'id' => $reservation->id,
+            'term'=> new TermResource($reservation->term),
+            'member_all' => $reservation->member_count,
+            'member_checked_in' => 0,
+        ]);
+    }
+
+    /**
+     * /reservation/{id}/check:get
+     */
+    public function testCheck() {
+        $user = User::factory()->permission('reservation')->create();
```

Comment:
実運用に即したい
どっちもチェックしてもいいかも

```suggestion
        $user = User::factory()->permission('executive')->create();
```